app/src/main/java/com/pusher/demo/features/marketplace/seller/SellerPresenter.kt

```diff
@@ -36,59 +40,68 @@ class SellerPresenter :  BasePresenter<SellerPresenter.View>(){
             })
     }
 
-    fun subscribeToRoom(room: Room) {
+    fun endSubscriptionToRoomUpdates() {
+        ChatkitManager.removeChatListener(chatListeners)
+    }
+
+    fun subscribeToRoomUpdates() {
```

Comment:
I think it would be simpler to set the listener already for the connect call and then have this fun as simply `fetchBuyersSubscribingToTheirPresence`. Also simplifying: we don't need a set of listeners but just one and don't need to store them but just pass them to the SDK on connect (onDestroy we could simply call `close` on `ChatManager` — which is more correct than just removing listeners on our side). That would simplify the things as now this function mixes a few responsibilities and it's name sounds like it's about room subscription (messages/typing indicators), not the high-level view of the chat.

Reply:
This `fun` also does not need to be called by the view. It can be internally called from within the presenter upon successful connect.